bundles/org.openhab.core.semantics/src/main/java/org/openhab/core/semantics/internal/SemanticsMetadataProvider.java

```diff
@@ -242,7 +246,6 @@ public void added(Item item) {
         processItem(item);
     }
 
-    @SuppressWarnings("null")
     @Override
     public void removed(Item item) {
         Metadata removedMd = semantics.remove(item.getName());
```

Comment:
This part of code can be replaced by `processItem(item)`.

Reply:
That would mean that I need to modify the item by removing the tags - otherwise the metadata is not removed since the removed item possibly still has a tag. I prefer to keep this untouched. :)

Reply:
Ah, got it. Thanks for clarification.